### ISSUE TITLE
building wheels?

### ISSUE THREAD

**Issue by xoviat** [SHTOOLS/SHTOOLS#100]
Would you mind forking the https://github.com/MacPython/scipy-wheels.git to this organization, renaming it to shtools-wheels, and adding Travis CI and appveyor to that repository? I can build wheels for Windows, Linux, and OSX.

**Comment by MarkWieczorek** [comment]
Ok, I forked and renamed the repo. I'll let @MMesch take care of Travis/appveyor since I don't know that much about it. Or I could give you write permissions for that repo if you want.

**Comment by xoviat** [comment]
I need to write permissions anyway.

**Comment by xoviat** [comment]
I don't have permission to enable the repository. Someone else will need to do that.

**Comment by xoviat** [comment]
Also, appveyor is a pain to set up for organizations.

**Comment by MarkWieczorek** [comment]
You should have write access for the build-shtools repo. What else do you need access to?

**Comment by xoviat** [comment]
Travis won't let me enable it.

**Comment by MMesch** [comment]
I am on a field trip right now, only occasional internet access. We need to activate travis for the build-shtools repository, correct?

**Comment by xoviat** [comment]
I think you have to be an admin to enable repositories on Travis which is not my role in this organization.

**Comment by MMesch** [comment]
I think it is activated now. Do you need anything else?

**Comment by xoviat** [comment]
Ideally appveyor as well, but that may be more difficult to set up.

**Comment by xoviat** [comment]
Manylinux is done.

**Comment by xoviat** [comment]
OSX is done. Win32 is probably a month away at least though. I need to fix some upstream problems (specifically, get flang working with numpy on windows) to make fortran compilation on WIN32 a more smooth process.

**Comment by xoviat** [comment]
Hopefully appveyor will be enabled by then as well.

**Comment by MarkWieczorek** [comment]
Great! Could you let me know how our release procedure needs to be updated using wheels? (never done this before) I'm thinking of putting out a new release in about a month's time.

**Comment by xoviat** [comment]
Just upload them to pypi and then people won't have to compile. Scipy uploads them to a rackspace container first but if you don't have access to cloud storage to get the wheels off of Travis, then you could upload them directly to pypi. You'll need to add some kind of private key or password to travis.yml with secure variables though.

**Comment by MarkWieczorek** [comment]
I think that AppVeyor is setup (the last build obviously failed). Let me know if it's not configured correctly.

**Comment by xoviat** [comment]
Update: the fortran compiler has been completed. Now I think we're just waiting on vcpkg integration.

**Comment by xoviat** [comment]
Actually I think we're ready to go. Let's do this.

**Comment by xoviat** [comment]
First attempt failed. It's looking for the missing fortran symbols:

```

  SHTOOLS.lib(MakeGridDH.o) : error LNK2001: unresolved external symbol dfftw_execute_
  SHTOOLS.lib(SHExpandDHC.o) : error LNK2001: unresolved external symbol dfftw_execute_
  SHTOOLS.lib(MakeGridDHC.o) : error LNK2001: unresolved external symbol dfftw_execute_
  SHTOOLS.lib(MakeGravGridDH.o) : error LNK2001: unresolved external symbol dfftw_destroy_plan_
  SHTOOLS.lib(MakeGravGradGridDH.o) : error LNK2001: unresolved external symbol dfftw_destroy_plan_
  SHTOOLS.lib(MakeMagGridDH.o) : error LNK2001: unresolved external symbol dfftw_destroy_plan_
  SHTOOLS.lib(SHExpandGLQ.o) : error LNK2001: unresolved external symbol dfftw_destroy_plan_
  SHTOOLS.lib(MakeGridGLQ.o) : error LNK2001: unresolved external symbol dfftw_destroy_plan_
  SHTOOLS.lib(SHExpandGLQC.o) : error LNK2001: unresolved external symbol dfftw_destroy_plan_
  SHTOOLS.lib(MakeGridGLQC.o) : error LNK2001: unresolved external symbol dfftw_destroy_plan_
  SHTOOLS.lib(SHExpandDH.o) : error LNK2001: unresolved external symbol dfftw_destroy_plan_
  SHTOOLS.lib(MakeGridDH.o) : error LNK2001: unresolved external symbol dfftw_destroy_plan_
```

**Comment by xoviat** [comment]
Windows is done. Though I could not test it.

**Comment by MarkWieczorek** [comment]
For info, I am going to start putting everything together for a release starting Nov 20 (I have no free time before that...) You might want to join our gitter; I'll ask everyone to take a look at your work.

**Comment by xoviat** [comment]
I asked on gitter, but what is left to do for the release other than uploading the wheels?

**Comment by MarkWieczorek** [comment]
I'm getting this error when trying to import the 4.1 osx wheel
````
In [1]: import pyshtools
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
RuntimeError: module compiled against API version 0xb but this version of numpy is 0xa
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-78d4dc3aa614> in <module>()
----> 1 import pyshtools

/usr/local/lib/python3.6/site-packages/pyshtools/__init__.py in <module>()
     45 
     46 # ---- Import shtools subpackages ----
---> 47 from . import shtools
     48 from . import constant
     49 from . import shclasses

/usr/local/lib/python3.6/site-packages/pyshtools/shtools/__init__.py in <module>()
     13 
     14 # legendre
---> 15 from .._SHTOOLS import PlmBar
     16 from .._SHTOOLS import PlmBar_d1
     17 from .._SHTOOLS import PlBar

ImportError: numpy.core.multiarray failed to import
````

**Comment by xoviat** [comment]
Addressed by #108.

**Comment by MarkWieczorek** [comment]
Everything looks good so far (v4.1.2), so I'm going to close this issue. Thanks for implementing this!
